src/request.rs

```diff
@@ -122,7 +122,7 @@ impl SecretType for SecretKey {
     async fn payload(&self, db: &KbsDb, connection: Connection) -> Result<Vec<u8>> {
         let key = db.get_secret(&self.request.id).await?;
         Ok(match &self.request.format[..] {
-            "binary" => key.into_bytes(),
+            "binary" => key.bytes()?,
             "json" => serde_json::to_string(&key).unwrap().into_bytes(),
```

Comment:
1. we should tackle at this `unwrap()` as well -- if there's invalid JSON string in the database, this `unwrap()` will panic the entire server.
2. we should add a test which populates invalid values into the DB and check that error is returned.

(both can wait for another PR)

Reply:
Yeah, I will do a follow-up addressing some other unwraps and a few other cleanup things I noticed .